builtin/v10/migration/top.go

```diff
@@ -75,25 +77,45 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, newManifestCID
 	// Set of prior version code CIDs for actors to defer during iteration, for explicit migration afterwards.
 	deferredCodeIDs := make(map[cid.Cid]struct{})
 
+	oldInitCodeCID := cid.Undef
 	for _, oldEntry := range oldManifestData.Entries {
 		newCodeCID, ok := newManifest.Get(oldEntry.Name)
 		if !ok {
 			return cid.Undef, xerrors.Errorf("code cid for %s actor not found in new manifest", oldEntry.Name)
 		}
 
 		migrations[oldEntry.Code] = migration.CodeMigrator{OutCodeCID: newCodeCID}
+
+		if oldEntry.Name == manifest.InitKey {
+			oldInitCodeCID = oldEntry.Code
+		}
 	}
 
+	if !oldInitCodeCID.Defined() {
+		return cid.Undef, xerrors.New("didn't find init actor in old manifest")
+	}
 	// migrations that migrate both code and state, override entries in `migrations`
 
 	// The System Actor
-	newSystemCodeCID, ok := newManifest.Get("system")
+	newSystemCodeCID, ok := newManifest.Get(manifest.SystemKey)
 	if !ok {
-		return cid.Undef, xerrors.Errorf("code cid for system actor not found in manifest")
+		return cid.Undef, xerrors.Errorf("code cid for system actor not found in new manifest")
 	}
 
 	migrations[systemActor.Code] = systemActorMigrator{OutCodeCID: newSystemCodeCID, ManifestData: newManifest.Data}
 
+	// The Init Actor
+	newInitCodeCID, ok := newManifest.Get(manifest.InitKey)
+	if !ok {
+		return cid.Undef, xerrors.Errorf("code cid for init actor not found in new manifest")
+	}
+
+	ethZeroAddr, err := makeEthZeroAddress()
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("failed to make eth zero address: %w", err)
+	}
+
+	migrations[oldInitCodeCID] = initActorMigrator{OutCodeCID: newInitCodeCID, EthZeroAddress: ethZeroAddr}
```

Comment:
Oh, I see. Yeah.... that's not so fun. Could we just apply it as an implicit message? Or is that not allowed.

Reply:
Fair enough.